be2-scala/src/main/scala/ch/epfl/pop/decentralized/Monitor.scala

```diff
@@ -51,7 +58,16 @@ final case class Monitor(
     case Monitor.TriggerHeartbeat =>
       log.info("triggering a heartbeat")
       timers.cancel(singleHbKey)
-      heartbeatGenRef ! Monitor.GenerateAndSendHeartbeat(connectionMediatorRef)
+
+      val askForHeartbeat = dbActorRef ? DbActor.GenerateHeartbeat()
+      val heartbeat: HashMap[Channel, Set[Hash]] =
+        Await.ready(askForHeartbeat, duration).value.get match
+          case Success(DbActor.DbActorGenerateHeartbeatAck(Some(map))) => map
+          case Failure(ex: DbActorNAckException)                       => HashMap.empty[Channel, Set[Hash]] // Specific failure
```

Comment:
Why keep the specific failure if the handling is the same as the case _? Isn't it possible to combine both?